vm/src/vm/runners/cairo_pie.rs

```diff
@@ -64,20 +81,225 @@ pub enum BuiltinAdditionalData {
     Output(OutputBuiltinAdditionalData),
     // Signatures are composed of (r, s) tuples
     #[serde(serialize_with = "serde_impl::serialize_signature_additional_data")]
-    Signature(HashMap<Relocatable, (Felt252, Felt252)>),
+    Signature(SignatureBuiltinAdditionalData),
```

Comment:
Is adding this struct needed? We can just implement a custom deserialization for it with #[serde(deserialize_with)] similar to how the custom serialization is implemented.

Reply:
This allows to deserialize the additional data from either a hashmap or a vector of tuples. This is easily implemented using `serde`'s visitor pattern, it's more of a pain to implement with a deserialization function.